File: data/transcoder_evaluation_gfg/go/MINIMUM_LENGTH_SUBARRAY_SUM_GREATER_GIVEN_VALUE_1.go
package main

import (
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(arr []int, n int, x int) int {
	var (
		curr_sum int = 0
		min_len  int = n + 1
		start    int = 0
		end      int = 0
	)
	for end < n {
		for curr_sum <= x && end < n {
			if curr_sum <= 0 && x > 0 {
				start = end
				curr_sum = 0
			}
			curr_sum += arr[func() int {
				p := &end
				x := *p
				*p++
				return x
			}()]
		}
		for curr_sum > x && start < n {
			if end-start < min_len {
				min_len = end - start
			}
			curr_sum -= arr[func() int {
				p := &start
				x := *p
				*p++
				return x
			}()]
		}
	}
	return min_len
}
//TOFILL
func main() {
	var (
		n_success int      = 0
		param0_0  []int  = []int{2, 4, 5, 10, 14, 15, 16, 20, 23, 28, 31, 35, 36, 36, 43, 48, 49, 55, 57, 57, 58, 61, 64, 64, 68, 70, 70, 73, 74, 76, 76, 77, 81, 81, 82, 87, 89, 92, 99}
		param0_1  []int  = []int{66, -20, 12, -48, 22, 28, 40, -30, -6, -96, 10, -88, 40}
		param0_2  []int  = []int{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}
		param0_3  []int  = []int{28, 19, 38, 31, 17, 27, 60, 35, 19, 47, 34, 51, 3, 95, 33, 29, 84, 46, 74, 87}
		param0_4  []int   = []int{-48, -2}
		param0_5  []int  = []int{1, 1, 0, 1, 1, 1, 1, 0, 1, 0, 1, 1, 0, 1, 1, 0, 1, 1, 0, 1, 1, 1}
		param0_6  []int  = []int{1, 4, 4, 6, 8, 10, 12, 12, 13, 15, 18, 20, 21, 23, 25, 28, 28, 33, 33, 35, 35, 36, 37, 38, 42, 44, 63, 63, 65, 65, 65, 66, 70, 74, 77, 78, 80, 80, 84, 87, 87, 89, 92, 93, 94, 97, 98, 99}
		param0_7  []int  = []int{-82, -12, -40, 58, 22, -76, -94, -28, 42, 36, 64}
		param0_8  []int  = []int{0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1}
		param0_9  []int  = []int{76, 65, 96, 76, 1, 91, 22, 29, 95, 21, 73, 68, 30, 52, 73, 86, 52, 66, 67, 37, 76, 53, 68, 6, 95, 81, 98, 42, 63, 38, 92, 78, 59, 86, 10, 38, 18, 15, 52, 62, 16, 66}
		param0    [][]int = [][]int{param0_0, param0_1, param0_2, param0_3, param0_4, param0_5, param0_6, param0_7, param0_8, param0_9}
		param1    []int  = []int{33, 11, 42, 15, 1, 12, 42, 10, 10, 23}
		param2    []int  = []int{28, 12, 23, 15, 1, 15, 27, 6, 14, 35}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if f_filled(([]int)(param0[i]), param1[i], param2[i]) == f_gold(([]int)(param0[i]), param1[i], param2[i]) {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}

File: data/transcoder_evaluation_gfg/rust/MINIMUM_LENGTH_SUBARRAY_SUM_GREATER_GIVEN_VALUE_1.rs
#![feature(main)]
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

#[no_mangle]
pub unsafe fn abs(mut x: i32)
 -> i32 {
    return x.abs();
}

#[no_mangle]
pub unsafe fn min(mut x: i32, mut y: i32)
 -> i32 {
    return if x < y { x } else { y };
}
#[no_mangle]
pub unsafe fn max(mut x: i32, mut y: i32)
 -> i32 {
    return if x > y { x } else { y };
}

pub unsafe fn f_gold(mut arr: *mut i32, mut n: i32,
                                mut x: i32) -> i32 {
    let mut curr_sum: i32 = 0 as i32;
    let mut min_len: i32 = n + 1 as i32;
    let mut start: i32 = 0 as i32;
    let mut end: i32 = 0 as i32;
    while end < n {
        while curr_sum <= x && end < n {
            if curr_sum <= 0 as i32 && x > 0 as i32 {
                start = end;
                curr_sum = 0 as i32
            }
            let fresh0 = end;
            end = end + 1;
            curr_sum += *arr.offset(fresh0 as isize)
        }
        while curr_sum > x && start < n {
            if end - start < min_len { min_len = end - start }
            let fresh1 = start;
            start = start + 1;
            curr_sum -= *arr.offset(fresh1 as isize)
        }
    }
    return min_len;
}
//TOFILL
unsafe fn main_0() -> i32 {
    let mut n_success: i32 = 0 as i32;
    let mut param0_0: [i32; 39] =
        [2 as i32, 4 as i32, 5 as i32,
         10 as i32, 14 as i32, 15 as i32,
         16 as i32, 20 as i32, 23 as i32,
         28 as i32, 31 as i32, 35 as i32,
         36 as i32, 36 as i32, 43 as i32,
         48 as i32, 49 as i32, 55 as i32,
         57 as i32, 57 as i32, 58 as i32,
         61 as i32, 64 as i32, 64 as i32,
         68 as i32, 70 as i32, 70 as i32,
         73 as i32, 74 as i32, 76 as i32,
         76 as i32, 77 as i32, 81 as i32,
         81 as i32, 82 as i32, 87 as i32,
         89 as i32, 92 as i32, 99 as i32];
    let mut param0_1: [i32; 13] =
        [66 as i32, -(20 as i32), 12 as i32,
         -(48 as i32), 22 as i32, 28 as i32,
         40 as i32, -(30 as i32), -(6 as i32),
         -(96 as i32), 10 as i32, -(88 as i32),
         40 as i32];
    let mut param0_2: [i32; 43] =
        [0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32];
    let mut param0_3: [i32; 20] =
        [28 as i32, 19 as i32, 38 as i32,
         31 as i32, 17 as i32, 27 as i32,
         60 as i32, 35 as i32, 19 as i32,
         47 as i32, 34 as i32, 51 as i32,
         3 as i32, 95 as i32, 33 as i32,
         29 as i32, 84 as i32, 46 as i32,
         74 as i32, 87 as i32];
    let mut param0_4: [i32; 2] =
        [-(48 as i32), -(2 as i32)];
    let mut param0_5: [i32; 22] =
        [1 as i32, 1 as i32, 0 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 0 as i32, 1 as i32,
         0 as i32, 1 as i32, 1 as i32,
         0 as i32, 1 as i32, 1 as i32,
         0 as i32, 1 as i32, 1 as i32,
         0 as i32, 1 as i32, 1 as i32,
         1 as i32];
    let mut param0_6: [i32; 48] =
        [1 as i32, 4 as i32, 4 as i32,
         6 as i32, 8 as i32, 10 as i32,
         12 as i32, 12 as i32, 13 as i32,
         15 as i32, 18 as i32, 20 as i32,
         21 as i32, 23 as i32, 25 as i32,
         28 as i32, 28 as i32, 33 as i32,
         33 as i32, 35 as i32, 35 as i32,
         36 as i32, 37 as i32, 38 as i32,
         42 as i32, 44 as i32, 63 as i32,
         63 as i32, 65 as i32, 65 as i32,
         65 as i32, 66 as i32, 70 as i32,
         74 as i32, 77 as i32, 78 as i32,
         80 as i32, 80 as i32, 84 as i32,
         87 as i32, 87 as i32, 89 as i32,
         92 as i32, 93 as i32, 94 as i32,
         97 as i32, 98 as i32, 99 as i32];
    let mut param0_7: [i32; 11] =
        [-(82 as i32), -(12 as i32), -(40 as i32),
         58 as i32, 22 as i32, -(76 as i32),
         -(94 as i32), -(28 as i32), 42 as i32,
         36 as i32, 64 as i32];
    let mut param0_8: [i32; 18] =
        [0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32];
    let mut param0_9: [i32; 42] =
        [76 as i32, 65 as i32, 96 as i32,
         76 as i32, 1 as i32, 91 as i32,
         22 as i32, 29 as i32, 95 as i32,
         21 as i32, 73 as i32, 68 as i32,
         30 as i32, 52 as i32, 73 as i32,
         86 as i32, 52 as i32, 66 as i32,
         67 as i32, 37 as i32, 76 as i32,
         53 as i32, 68 as i32, 6 as i32,
         95 as i32, 81 as i32, 98 as i32,
         42 as i32, 63 as i32, 38 as i32,
         92 as i32, 78 as i32, 59 as i32,
         86 as i32, 10 as i32, 38 as i32,
         18 as i32, 15 as i32, 52 as i32,
         62 as i32, 16 as i32, 66 as i32];
    let mut param0: [*mut i32; 10] =
        [param0_0.as_mut_ptr(), param0_1.as_mut_ptr(), param0_2.as_mut_ptr(),
         param0_3.as_mut_ptr(), param0_4.as_mut_ptr(), param0_5.as_mut_ptr(),
         param0_6.as_mut_ptr(), param0_7.as_mut_ptr(), param0_8.as_mut_ptr(),
         param0_9.as_mut_ptr()];
    let mut param1: [i32; 10] =
        [33 as i32, 11 as i32, 42 as i32,
         15 as i32, 1 as i32, 12 as i32,
         42 as i32, 10 as i32, 10 as i32,
         23 as i32];
    let mut param2: [i32; 10] =
        [28 as i32, 12 as i32, 23 as i32,
         15 as i32, 1 as i32, 15 as i32,
         27 as i32, 6 as i32, 14 as i32,
         35 as i32];
    let mut i: i32 = 0 as i32;
    while i < param0.len() as i32 {
        if f_filled(param0[i as usize], param1[i as usize],
                    param2[i as usize]) ==
               f_gold(param0[i as usize], param1[i as usize],
                      param2[i as usize]) {
            n_success += 1 as i32
        }
        i += 1
    }
    println!("{} {} {} {} {}", "#Results:", " ", n_success, ", ", param0.len() as i32);
    return 0 as i32;
}
#[main]
pub fn main() { unsafe { ::std::process::exit(main_0() as i32) } }
